cocos/2d/CCSprite.cpp

```diff
@@ -373,9 +373,12 @@ void Sprite::setTexture(const std::string &filename)
 
 void Sprite::setTexture(Texture2D *texture)
 {
+    //onetime conversion to std::string since GLProgramState::getOrCreateWithGLProgramName takes a std::string instead of const char*
+    static std::string SHADER_NAME_POSITION_TEXTURE_COLOR_NO_MVP = GLProgram::SHADER_NAME_POSITION_TEXTURE_COLOR_NO_MVP;
+
     if(_glProgramState == nullptr)
     {
-        setGLProgramState(GLProgramState::getOrCreateWithGLProgramName(GLProgram::SHADER_NAME_POSITION_TEXTURE_COLOR_NO_MVP, texture));
```

Comment:
many codes use the `GLProgram::SHADER_NAM...` like this, and then cause another translation from `const char*` to `std::string`.

To correct this issue, I think it's more proper to modify the `GLProgram::SHADER...` definition on `CCGLProgram.cpp`

Reply:
Do you mean that they should all be made `std::string` instead? I haven't looked for the conversions in the opposite direction, but I'll take a look and see what's going on.

Reply:
@drelaptop does it matter that much translation from `const char* to std::string`. Just wondering?

Reply:
yes, I mean all `std::string` instead, for those are same issue with this.

> the main aim is keep uniform for same usage on similar things. change all is better than change none, than change one of them.

Reply:
May be, in the future(C++17 or later): we should use std::string_view to instead ```const st::string&``` and ```const char*``` as parameter, see https://github.com/halx99/mini-asio/blob/master/src/string_view.hpp